packages/vector_graphics_compiler/lib/src/svg/node.dart

```diff
@@ -15,8 +15,8 @@ typedef Resolver<T> = T Function(String id);
 /// Nodes describe painting attributes, clips, transformations, paths, and
 /// vertices to draw in depth-first order.
 abstract class Node {
-  /// Allows subclasses to be const.
-  const Node();
+  /// This node's parent, or `null` if it is the root element.
+  Node? parent;
```

Comment:
Please no.

Let's just pass the information needed down to children as needed. I really really really want to avoid having mutable nodes. It makes it more difficult to reason about what has happened to a node when you pass it to a function.

Reply:
So I need this information to eventually implement dx/dy support for text. i.e. I want to ask the question "what was the previous text node in the tree". Doing that without any sort of mutation requires uss to build a cache as we go, which is definitely feasible - but I think in that case we shouldn't do it all in a single build method - instead we could have a separate resolve/build and use the former to update the tree and convert all relative sizes to absolute. That would require more passes but since we're doing this offline it may not matter.

Thoughts?

Reply:
In flutter_svg this gets built up at parse time becuase it actually creates ui.Paragraph objects and can convert relative to absolute.

We're not going to be able to do that in the compiler now becuase we don't know glyph sizes for the font that resolves on the end device.

I think we'll have to just pass in relative units to the runtime so that it can figure out glyph positions for us.

That said, I think we should hold off on aggressively implementing this for now, since its not really anything that customer money needs at this point.

Reply:
IOW I'm pretty sure having the parent here won't help you, because you don't know how to calculate the current text position that dx/dy would refer to, until there are actual paragraph objects laid out at runtime.

Reply:
My first read of the spec was that it was relative to that text's baseline which we do know, but you might be right. Agreed on holding off on this for now.

I removed the mutability and pass down the parent Rect through build

Reply:
Spec says

> Shifts in the [current text position](https://www.w3.org/TR/SVG11/text.html#CurrentTextPosition) along the x-axis for the characters within this element or any of its descendants.

Curent text position is documented as involving knowing glyph size and spacing.

Reply:
ewwwwwwww